### ISSUE TITLE
Regression in startup time for EDL files with many chapters

### ISSUE THREAD

**Issue by xM8WVqaG** [mpv-player/mpv#7267]
tl;dr, mpv used to load an EDL with many chapters in it that references a large file really fast and it now it's much slower.

### mpv version and platform

Windows 7 64-bit

Previous mpv Version

```
C:\debug>mpv-x86_64-20180731\mpv.com --version
mpv 0.29.0 Copyright © 2000-2018 mpv/MPlayer/mplayer2 projects
 built on UNKNOWN
ffmpeg library versions:
   libavutil       56.18.102
   libavcodec      58.22.100
   libavformat     58.17.101
   libswscale      5.2.100
   libavfilter     7.26.100
   libswresample   3.2.100
ffmpeg version: 4.1+809.g269daf5985
```

Current latest shinchiro Version

```
C:\debug>mpv-x86_64-20191215-git-2dbe33c\mpv.com --version
mpv 0.30.0-302-g2dbe33ce83 Copyright © 2000-2019 mpv/MPlayer/mplayer2 projects
 built on Sun Dec 15 17:01:14 +08 2019
ffmpeg library versions:
   libavutil       56.36.101
   libavcodec      58.65.100
   libavformat     58.35.101
   libswscale      5.6.100
   libavfilter     7.68.101
   libswresample   3.6.100
ffmpeg version: git-2019-12-15-9fe07908
```

### Reproduction steps

The example below exaggerates the problem, but the problem is still very noticeable after only five or six chapters.

#### Create a large EDL

To generate an EDL with 100 chapters:

```bash
#!/bin/bash
i=0
duration=60

echo \# mpv EDL v0
while [ $i -lt 100 ]
do
  echo sample.mp4,$((i * duration)),$duration
  i=$((i + 1))
done
```

```
$ ./generate-edl.sh > sample.edl
```

#### Load it into mpv

```
C:\debug>mpv-x86_64-20180731\mpv.com --no-config --log-file=20180731.log sample\foo.edl
```

Playback began in less than 500 ms.

Logs: https://0x0.st/z0bA.log

```
C:\debug>mpv-x86_64-20191215-git-2dbe33c\mpv.com --no-config --log-file=20191215.log sample\foo.edl
```

Playback began in about 217 seconds.

Logs: https://0x0.st/z0bm.log

### Expected behavior

mpv should load and start playing the file at a similar time to other files.

### Actual behavior

mpv starts playing the EDL several seconds later depending on the amount of cuts and size of file.

### Log file

_See reproduction steps for logs._

### Sample files

_This can be reproduced without any specific files._

**Comment by xM8WVqaG** [comment]
@wm4 Thanks for the rapid turn around! :heart:
